sdk/messaging/azeventhubs/internal/rpc.go

```diff
@@ -101,7 +91,7 @@ func NewRPCLink(ctx context.Context, args RPCLinkArgs) (*rpcLink, error) {
 
 	linkID, err := uuid.New()
 	if err != nil {
-		closeOrLog("session", session)
+		_ = session.Close(ctx)
```

Comment:
`Close()` could return an "interesting" error, e.g. if the parent `Conn` died during closing.  Maybe you want to defer handling that until the next time `Conn` is used?  At any rate, just pointing it out.

Reply:
My thinking was along the lines of your second point - that it'd just be more descriptive the next time we attempt to use the Conn, so we didn't have to get into too complicated an error return strategy here.